tests/Unit/Elliptic/Systems/Poisson/Actions/Test_Observe.cpp

```diff
@@ -0,0 +1,302 @@
+#include "IO/H5/VolumeData.hpp"
+#include "IO/Observer/Actions.hpp"  // IWYU pragma: keep
+#include "IO/Observer/Initialize.hpp"
+#include "IO/Observer/ObserverComponent.hpp"
+#include "IO/Observer/Tags.hpp"
+#include "IO/Observer/TypeOfObservation.hpp"
+#include "NumericalAlgorithms/LinearSolver/IterationId.hpp"
+#include "NumericalAlgorithms/LinearSolver/Tags.hpp"  // IWYU pragma: keep
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Parallel/Reduction.hpp"
+#include "Utilities/Algorithm.hpp"
+#include "Utilities/FileSystem.hpp"
+#include "Utilities/Functional.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+#include "tests/Unit/ActionTesting.hpp"
+
+namespace PUP {
+class er;
+}  // namespace PUP
+// IWYU pragma: no_forward_declare Tensor
+
+namespace {
+template <typename Metavariables>
+struct MockElementArray {
+  using component_being_mocked = void;  // Not needed
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = ElementIndex<2>;
+  using const_global_cache_tag_list = tmpl::list<>;
+  using action_list = tmpl::list<Poisson::Actions::Observe>;
+  using simple_tags =
+      db::AddSimpleTags<LinearSolver::Tags::IterationId, Tags::Mesh<2>,
+                        Poisson::Field, Tags::Coordinates<2, Frame::Inertial>>;
+  using compute_tags = db::AddComputeTags<>;
+  using initial_databox =
+      db::compute_databox_type<tmpl::append<simple_tags, compute_tags>>;
+};
+
+template <typename Metavariables>
+struct MockObserverComponent {
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = size_t;
+  using const_global_cache_tag_list = tmpl::list<>;
+  using action_list = tmpl::list<>;
+  using component_being_mocked = observers::Observer<Metavariables>;
+  using simple_tags =
+      typename observers::Actions::Initialize<Metavariables>::simple_tags;
+  using compute_tags =
+      typename observers::Actions::Initialize<Metavariables>::compute_tags;
+  using initial_databox =
+      db::compute_databox_type<tmpl::append<simple_tags, compute_tags>>;
+};
+
+template <typename Metavariables>
+struct MockObserverWriterComponent {
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = size_t;
+  using const_global_cache_tag_list =
+      tmpl::list<observers::OptionTags::ReductionFileName,
+                 observers::OptionTags::VolumeFileName>;
+  using action_list = tmpl::list<>;
+  using component_being_mocked = observers::ObserverWriter<Metavariables>;
+  using simple_tags =
+      typename observers::Actions::InitializeWriter<Metavariables>::simple_tags;
+  using compute_tags = typename observers::Actions::InitializeWriter<
+      Metavariables>::compute_tags;
+  using initial_databox =
+      db::compute_databox_type<tmpl::append<simple_tags, compute_tags>>;
+};
+
+struct AnalyticSolution {
```

Comment:
it seems undesirable in terms of testing robustness to have the `AnalyticSolution` you're writing be identical to the also written x-coordinate value. You could imagine a bug where the incorrect thing is written to or retrieved from the file that this wouldn't disambiguate. A simple fix might be to just perform some simple mathematical operation here to make sure that you get different values in each slot, and update the test below.

Reply:
you're right, fixed it